surface/include/pcl/surface/3rdparty/opennurbs/opennurbs_array.h

```diff
@@ -349,8 +349,8 @@ template <class T> class ON_SimpleArray
   // implimentation //////////////////////////////////////////////////////
   void Move( int /* dest index*/, int /* src index */, int /* element count*/ );
 	T*   m_a;        // pointer to array memory
-	int  m_count;    // 0 <= m_count <= m_capacity
-	int  m_capacity; // actual length of m_a[]
+	unsigned int  m_count;    // 0 <= m_count <= m_capacity
+	unsigned int  m_capacity; // actual length of m_a[]
```

Comment:
This is a potentially dangerous change. You're implicitly relying on the original author not using negative numbers for status checks. Any idea how many warnings this was generating before? How feasible is it to apply casts solely on the locations in which warnings were triggered?

Reply:
The array_defs.h file shows extensive checks for negative values and casts to `unsigned`

Reply:
My issue is that "extensive" does not imply "complete". If a single case is missed then we potentially introduced a silent issue. Hence, my original question, how feasible is it to just address the warnings individually?

Reply:
I understand it could quite possibly lead to some silent error. The author should have used unsigned types from the beginning, which would prevent corner cases and reduced the number of checks required, but anyway that's not the point here I guess.
Have applied casts to where it was needed only now.